core/src/queue.rs

```diff
@@ -137,59 +145,88 @@ impl Queue {
     ///
     /// # Errors
     /// See [`enum@Error`]
-    #[allow(
-        clippy::unwrap_in_result,
-        clippy::expect_used,
-        clippy::missing_panics_doc
-    )]
     pub fn push(
         &self,
         tx: VersionedAcceptedTransaction,
         wsv: &WorldStateView,
-    ) -> Result<(), (VersionedAcceptedTransaction, Error)> {
-        if tx.is_in_future(self.future_threshold) {
-            Err((tx, Error::InFuture))
-        } else if let Err(e) = self.check_tx(&tx, wsv) {
-            Err((tx, e))
-        } else if self.txs.len() >= self.max_txs {
-            Err((tx, Error::Full))
-        } else {
-            let hash = tx.hash();
-            let entry = match self.txs.entry(hash) {
-                Entry::Occupied(mut old_tx) => {
-                    // MST case
-                    old_tx
-                        .get_mut()
-                        .as_mut_v1()
-                        .signatures
-                        .extend(tx.as_v1().signatures.clone());
-                    return Ok(());
+    ) -> Result<(), Failure> {
+        match self.check_tx(&tx, wsv) {
+            Err(err) => Err(Failure { tx, err }),
```

Comment:
I think we should remove transaction from the queue here and maybe in `is_pending` as well

Reply:
hmm, but it's `push` method, so transaction is not in the queue yet

Reply:
ok, then in the `is_pending` we should remove transactions?

Reply:
`is_pending` is predicate function why we should remove transactions inside it?

Reply:
ok

Reply:
> is_pending is predicate function why we should remove transactions inside it?

is `is_pending` is also used in `all_transactions` but we don't remove from the queue

Reply:
This doesn't give much. Removing from map will not really free space for new transactions, because queues still will be occupied with hashes of removed transactions.